h2o-core/src/main/java/hex/AUC2.java

```diff
@@ -349,46 +396,36 @@ private void mergeOneBin( ) {
       System.arraycopy(_sqe,ssx+2,_sqe,ssx+1,_n-ssx-2);
       System.arraycopy(_tps,ssx+2,_tps,ssx+1,_n-ssx-2);
       System.arraycopy(_fps,ssx+2,_fps,ssx+1,_n-ssx-2);
+      System.arraycopy(_sqeEst, ssx+2, _sqeEst, ssx+1, _n-ssx-2);
       _n--;
-      _ssx = -1;
+      _ssx = -1;   // reset so that the next mergeOneBin() can start over
+      updateSqeEst(ssx, _n);
+      return ssx;
     }
 
-    // Find the pair of bins that when combined give the smallest increase in
-    // squared error.  Dups never increase squared error.
-    //
-    // I tried code for merging bins with keeping the bins balanced in size,
-    // but this leads to bad errors if the probabilities are sorted.  Also
-    // tried the original: merge bins with the least distance between bin
-    // centers.  Same problem for sorted data.
     private int find_smallest() {
-      if( _ssx == -1 ) return (_ssx = find_smallest_impl());
-      assert _ssx == find_smallest_impl();
-      return _ssx;
-    }
-    private int find_smallest_impl() {
       double minSQE = Double.MAX_VALUE;
       int minI = -1;
       int n = _n;
       for( int i=0; i<n-1; i++ ) {
-        double derr = compute_delta_error(_ths[i+1],k(i+1),_ths[i],k(i));
-        if( derr == 0 ) return i; // Dup; no increase in SQE so return immediately
-        double sqe = _sqe[i]+_sqe[i+1]+derr;
-        if( sqe < minSQE ) {
-          minI = i;  minSQE = sqe;
+        if( _sqeEst[i] < minSQE ) {
```

Comment:
Is the only purpose of `_sqeEst` to avoid the calculation on lines 373-374?

Can you please explain the purpose of it?